### PR TITLE
[jit] Clean up old uses of checkScript

### PR DESCRIPTION
**Pull request by driazati**
This cleans up the `checkScript` API and some old tests that were hardcoding outputs. It also now runs the Python function when a string is passed in to verify the outputs.

Differential Revision: [D15913034](https://our.internmc.facebook.com/intern/diff/15913034/)

### CODE REVIEW

File: test/jit_utils.py [hunk 1]
```diff
@@ -271,51 +280,65 @@ def checkScriptRaisesRegex(self, script, inputs, exception, regex,
             ge = torch.jit.script(script, optimize)
             ge(*inputs)
 
+    def get_frame_vars(self, frames_up):
```
Comment (eellison) [inline test/jit_utils.py]: Why is this necessary to add?

Comment (driazati) [reply test/jit_utils.py]: To `exec` the function you need to pass the scope so it can lookup values

File: test/test_jit.py [hunk 2]
```diff
@@ -5121,7 +5115,7 @@ def func_float_int(a, b):
         def divmod_test_iterator(func, num, den):
             for i in num:
                 for j in den:
-                    self.checkScript(func, (i, j))
+                    self.checkScript(func, (i, j), frames_up=2)
```
Comment (eellison) [inline test/test_jit.py]: It's odd that this now has to manually pass in the frame when it didn't before. Was the default 2 prior and now it's 1?

Comment (driazati) [reply test/test_jit.py]: It's nested in another function so we need to go up an extra frame to evaluate the function with the string frontend (which entails creating a Python function from a code string, which needs to look at the stack to get the locals/globals/closed over vars)

File: test/test_jit.py [hunk 3]
```diff
@@ -5478,19 +5465,14 @@ def throwsAnd(t):
...
+        def func(v):
             # type: ({from_type}) -> {to_type}
             return {to_type}(v)
         ''')
 
         def check_cast(from_type, to_type, value, raises=False):
             code = template.format(from_type=from_type, to_type=to_type)
-            expected = getattr(builtins, to_type)(value)
-            if raises:
-                with self.assertRaisesRegex(RuntimeError, "Cannot cast"):
```
Comment (eellison) [inline test/test_jit.py]: What happened to the assertRaisesRegex here ?

Comment (driazati) [reply test/test_jit.py]: Dead code

**Review by eellison — APPROVED** [review]
LGTM
